frontend/src/modules/editor/themeEditor/views/editorThemingView.js

```diff
@@ -0,0 +1,546 @@
+
+      EditorOriginView.prototype.initialize.apply(this, arguments);
+    },
+
+    preRender: function() {
+      this.$el.hide();
+    },
+
+    render: function() {
+      EditorOriginView.prototype.render.apply(this, arguments);
+
+      Origin.trigger('location:title:update', {
+        breadcrumbs: ['dashboard','course', { title: Origin.l10n.t('app.themeeditor') }],
+        title: Origin.l10n.t('app.themingtitle')
+      });
+
+      this.updateRestorePresetButton();
+      this.renderForm();
+    },
+
+    renderForm: function() {
+      this.removeForm();
+
+      var selectedTheme = this.getSelectedTheme();
+
+      if (!this.themeIsEditable(selectedTheme)) {
+        this.$('.tile.preset').hide();
+        this.$('.edit.btn.secondary').hide();
+        return;
+      }
+
+      this.$('.tile.preset').show();
+      this.$('.edit.btn.secondary').show();
+      try {
+        this.form = Origin.scaffold.buildForm({
+          model: selectedTheme,
+          schemaType: selectedTheme.get('theme')
+        });
+      }
+      catch(e) {
+        console.log(e);
+      }
+
+      if (this.form) {
+        this.$('.form-container').html(this.form.el);
+        this.$('.empty-message').hide();
+      } else {
+        this.$('.empty-message').show();
+      }
+
+      this.$el.find('fieldset:not(:has(>.field))').addClass('empty-fieldset');
+      this.$('.theme-customiser').show();
+      Origin.trigger('theming:showPresetButton', true);
+
+      var toRestore = Origin.editor.data.course.get('themeVariables') || this.getDefaultThemeSettings();
+      _.defer(function() { this.restoreFormSettings(toRestore); }.bind(this));
+    },
+
+    removeForm: function() {
+      this.$('.form-container').empty();
+      this.$('.theme-customiser').hide();
+
+      this.form = null;
+
+      Origin.trigger('theming:showPresetButton', false);
+    },
+
+    postRender: function() {
+      this.updateSelects();
+      this.setViewToReady();
+
+      this.$el.show();
+    },
+
+    loadCollections: function() {
+      this.themes = new ThemeCollection();
+      this.listenTo(this.themes, {
+        'sync': this.onCollectionReady,
+        'error': this.onError
+      });
+      this.themes.fetch();
+
+      this.presets = new PresetCollection();
+      this.listenTo(this.presets, {
+        'sync': this.onCollectionReady,
+        'error': this.onError
+      });
+      this.presets.fetch();
+    },
+
+    updateSelects: function() {
+      this.updateThemeSelect();
+      this.updatePresetSelect();
+    },
+
+    updateThemeSelect: function() {
+      var select = this.$('.theme select');
+      var oldVal = select.val();
+      // remove options first
+      $('option', select).remove();
+      // add 'no presets'
+      select.append($('<option>', { value: "", disabled: 'disabled', selected: 'selected' }).text(Origin.l10n.t('app.selectinstr')));
+      // add options
+      _.each(this.themes.models, function(item, index) {
```

Comment:
Not sure whether we can be bothered to update but any `_.each()` loops on arrays can use the native `.forEach()`. 

Similarily, `_.keys()` → `Object.keys()`.

We only ever used the Underscore versions in Adapt for IE8(!).

Reply:
I've changed the ones in the themeEditor directory